### ISSUE TITLE
Two errors with angular 10 strict mode

### ISSUE THREAD

**Issue by samizarraa** [mzuccaroli/angular-google-tag-manager#40]
Hi
I got this errors with latest update:

> ERROR in node_modules/angular-google-tag-manager/lib/google-tag-manager-config.d.ts:3:5 - error TS2411: Property 'gtm_auth' of type 'string | undefined' is not assignable to string index type 'string'.

gtm_auth?: string;
      ~~~~~~~~

> node_modules/angular-google-tag-manager/lib/google-tag-manager-config.d.ts:4:5 - error TS2411: Property 'gtm_preview' of type 'string | undefined' is not assignable to string index type 'string'.

gtm_preview?: string;

"angular-google-tag-manager": "^1.2.1"
"@angular/core": "^10.1.5",

**Comment by mzuccaroli** [comment]
hi @samizarraa  I'm working on that
meanwhile I think the 1.2.0 version could fit the strict mode

@GravlLift , the problem is caused by the GoogleTagManagerConfig interface, any suggestion to fix it?

**Comment by GravlLift** [comment]
Indeed. If you wish to use strict mode in typescript, the `GoogleTagManagerConfig` interface needs to be defined like so:
```ts
export interface GoogleTagManagerConfig {
  id: string | null;
  gtm_auth?: string;
  gtm_preview?: string;
  [key: string]: string | null | undefined;
}
```

**Comment by samizarraa** [comment]
Than you

**Comment by mzuccaroli** [comment]
fixed in last release, update do 1.2.2 version
thanks @samizarraa  and @GravlLift

**Comment by samizarraa** [comment]
It works fine now, Thanks
